packages/design-system/src/components/MonthPicker/MonthPicker.tsx

```diff
@@ -0,0 +1,217 @@
+import Button, { ButtonVariation } from '../Button/Button';
+import Choice from '../ChoiceList/Choice';
+import React, { ChangeEvent, useState } from 'react';
+import classNames from 'classnames';
+import pick from 'lodash/pick';
+
+const monthNumbers = (() => {
+  const months = [];
+  for (let m = 1; m <= NUM_MONTHS; m++) {
+    months.push(m);
+  }
+  return months;
+})();
+
+export type MonthPickerErrorPlacement = 'top' | 'bottom';
+
+interface MonthPickerProps {
+  /**
+   * The `input` field's `name` attribute
+   */
+  name: string;
+  /**
+   * A [BCP 47 language tag](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#Locale_identification_and_negotiation)
+   * for month name localization. For example: Passing `es-US` as a value
+   * will render month names in Spanish.
+   */
+  locale?: string;
+  /**
+   * Additional classes to be added to the root element.
+   */
+  className?: string;
+  /**
+   * Applies the "inverse" UI theme
+   */
+  inversed?: boolean;
+  /**
+   * Variation string to be applied to buttons. See [Button component]({{root}}/components/button/#components.button.react)
+   */
+  buttonVariation?: ButtonVariation;
+  /**
+   * Label for the field
+   */
+  label: React.ReactNode;
+  /**
+   * Enable the error state by providing an error message.
+   */
+  errorMessage?: React.ReactNode;
+  /**
+   * Additional classes to be added to the error message
+   */
+  errorMessageClassName?: string;
+  /**
+   * Location of the error message relative to the field input
+   */
+  errorPlacement?: MonthPickerErrorPlacement;
+  /**
+   * Additional hint text to display
+   */
+  hint?: React.ReactNode;
+  /**
+   * Text showing the requirement ("Required", "Optional", etc.). See [Required and Optional Fields]({{root}}/guidelines/forms/#required-and-optional-fields).
+   */
+  requirementLabel?: React.ReactNode;
+  /**
+   * Array of month numbers, where `1` is January, and any month included
+   * is disabled for selection.
+   */
+  disabledMonths?: number[];
+  /**
+   * Array of month numbers, where `1` is January, and any month included
+   * is selected. This will render a read-only field. If the field should
+   * be mutable, use `defaultSelectedMonths`.
+   */
+  selectedMonths?: number[];
+  /**
+   * Array of month numbers, where `1` is January, and any month included
+   * is selected by default. Sets the initial checked state for the 12 month
+   * checkboxes. Use this for an uncontrolled component; otherwise, use the
+   * `selectedMonths` property.
+   */
+  defaultSelectedMonths?: number[];
+  /**
+   * A callback function that's invoked when a month's checked state is changed.
+   * Note: This callback is not called when a month is selected or deselected
+   * via the "Select all" or "Clear all" buttons – use the `onSelectAll` and
+   * `onClearAll` event handlers for those instances.
+   */
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => any;
```

Comment:
Just a question - I think I've been setting my event types to `React.MouseEvent | React.KeyboardEvent`. What's the difference between that approach and what you're using? Does my approach have outages in what could be considered an event?

Reply:
As far as I know, this is the type of the event that gets fired. I'm not really sure where those two events you mentioned would get used—maybe if you're specifically trying to target events from certain input sources? I know that `React.ChangeEvent` has `target` and `currentTarget`, whereas there's a more generic `SyntheticEvent` that is only guaranteed to have `currentTarget` 🤷